trash: Create inode_table only while feature is enabled

mohit84 opened the pull request:
Currently trash xlator create a inode table(1M) even if
feature is not enabled.In brick_mux environment while 250
bricks are attached with a single brick process and feature
is not enable brick process increase RSS size unnecessarily.

Solution: Create inode_table only while a feature is enabled.
          The patch reduces 250M RSS size per brick process
          if trash feature is not enabled.

Change-Id: I11a6fd2b8419fe2988f398be6ec30fb4f3b99a5d
Fixes: #1543
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

mohit84 committed:
trash: Create inode_table only while feature is enabled

Currently trash xlator create a inode table(1M) even if
feature is not enabled.In brick_mux environment while 250
bricks are attached with a single brick process and feature
is not enable brick process increase RSS size unnecessarily.

Solution: Create inode_table only while a feature is enabled.
          The patch reduces 250M RSS size per brick process
          if trash feature is not enabled.

Change-Id: I11a6fd2b8419fe2988f398be6ec30fb4f3b99a5d
Fixes: #1543
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

mohit84 commented:
/run regression

amarts commented:
@deepshikhaaa Can you check if the regression is indeed running at all ? 

https://build.gluster.org/job/gh_centos7-regression/6/console

looks like something is missing there!

deepshikhaaa commented:
/run regression

mohit84 committed:
configure.ac: Add --enable-brickmux

Introduce an --enable-brickmux configure option to
enable brick_mux as a default option.

Fixes: #1569
Change-Id: Id0c61ee3500630889e2cb2b5816cd40f68d47c94
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

xhernandez commented:
It seems like the `--enable-brickmux` patch should go to another PR.

xhernandez commented on xlators/features/trash/src/trash.c:
```diff
@@ -2246,6 +2246,9 @@ reconfigure(xlator_t *this, dict_t *options)
 
     if (priv->state) {
+        if (!priv->trash_itable)
+            priv->trash_itable = inode_table_new(0, this);
```
Shouldn't we destroy the table if the option is disabled ? Otherwise we'll still waste memory, which I think it's what is trying to do this patch.

Has any side consequences destroying the table online ? if so, probably we shouldn't allow online modification of this option.

mohit84 commented:
The patch is already merged by different pull request so closing this.
